backend/backend.go

```diff
@@ -27,6 +29,22 @@ type Backend interface {
 	State() (state.State, error)
 }
 
+// MultiState is an interface that a backend can implement to allow changing
+// between named states depending on the configured environment.
+type MultiState interface {
+	// States returns a list of configured named states and the current state.
+	States() ([]string, string, error)
```

Comment:
I like to err on the side of smaller interfaces (isn't there some tao of Go quote about small interfaces? 😉 ) Though if we do want to enforce this it's not too unwieldy and not nearly the largest in terraform.